From 9304ad88a5f4b083d348563c5de00da53b34cf46 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Fri, 6 Mar 2020 23:20:39 +0100
Subject: [PATCH] Makefile.am: fix build without makeinfo

Don't build doc subdir if makeinfo is not available otherwise build
fails on:

make[4]: Entering directory '/nvmedata/autobuild/instance-7/output-1/build/host-guile-2.0.14/doc/ref'
  MAKEINFO guile.info
/nvmedata/autobuild/instance-7/output-1/build/host-guile-2.0.14/build-aux/missing: line 81: makeinfo: command not found
WARNING: 'makeinfo' is missing on your system.
         You should only need it if you modified a '.texi' file, or
         any other file indirectly affecting the aspect of the manual.
         You might want to install the Texinfo package:
         <http://www.gnu.org/software/texinfo/>
         The spurious makeinfo call might also be the consequence of
         using a buggy 'make' (AIX, DU, IRIX), in which case you might
         want to install GNU make:
         <http://www.gnu.org/software/make/>

Fixes:
 - http://autobuild.buildroot.org/results/9605aac6f760bfff190d0ab95fa50f65486ffe90

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: sent to bug-guile@gnu.org]
---
 Makefile.am | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index ebbf6d476..2270afb9f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -38,8 +38,12 @@ SUBDIRS =					\
 	test-suite				\
 	benchmark-suite				\
 	gc-benchmarks				\
-	am					\
+	am
+
+if HAVE_MAKEINFO
+SUBDIRS +=					\
 	doc
+endif
 
 DIST_SUBDIRS = $(SUBDIRS) prebuilt
 
-- 
2.25.0

